Yearn-agents/InstantWithdrawl/src/utils.ts

```diff
@@ -0,0 +1,73 @@
+import { Finding, FindingSeverity, FindingType } from "forta-agent";
+
+import { BigNumber } from "bignumber.js";
+
+export type UserDetails = {
+  account: string;
+  balance: BigNumber;
+};
+export type Mapping = {
+  [key: string]: Array<UserDetails>;
+};
+
+export const getAccounts = async (axios: any) => {
+  const query = `{
+    accountVaultPositions (orderBy:balancePosition, orderDirection:desc, first:150){
+      id, account{
+        id
+      },
+      balancePosition,
+      balanceShares,
+      vault{
+        id
+      }
+    }
+  }`;
+
+  var data = JSON.stringify({
+    query,
+  });
+
+  var config = {
+    method: "post",
+    url: "https://api.thegraph.com/subgraphs/name/salazarguille/yearn-vaults-v2-subgraph-mainnet",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    data,
+  };
+
+  const res = await axios(config as any);
+  const positions = res.data.data.accountVaultPositions;
+  const mapping: Mapping = {};
+  const users = new Set();
+
+  positions.forEach((value: any) => {
+    const address: string = value.vault.id;
+    const values = mapping[address] || [];
+    values.push({
+      account: value.account.id,
+      balance: new BigNumber(value.balanceShares),
+    });
+    users.add(address);
+    mapping[address] = values;
+  });
+  return { mapping, users };
+};
+
+export function generateFinding(
+  balance: string,
+  index: number,
+  vaultAddress: string
+) {
+  return Finding.fromObject({
+    name: "Yearn-agent-7",
+    description: `The ${index} account wasn't able to make the withdraw on vault ${vaultAddress}`,
+    severity: FindingSeverity.Info,
+    type: FindingType.Unknown,
+    alertId: "Yearn-agent-7",
+    metadata: {
+      balance,
+    },
+  });
```

Comment:
Please use the `address` instead of the `index`. Also make the description simpler by moving `vaultAddress` and the `address` to the metadata

Reply:
I can move the vaultAddress to metadata for sure, but cant use address, the requirements need the number of top balance holders that can withdraw not the addresses itself. They want the index.

Reply:
Well, you can use both if the index is needed. Please add the `address` too.

Reply:
No. Read the requirements first, please. We have to report the number of addresses that can withdraw, not the address. Its redundant information only, and is not needed at all.